tests/test-dladdr.c

```diff
@@ -0,0 +1,180 @@
+    if( verbose || !passed )
+    {
+        if( !result )
+            printf( "(could not get symbol information for address %p)\n", addr );
+        else
+            print_dl_info( &info, "(", ")\n");
+    }
+    return !passed;
+}
+
+#ifdef _WIN32
+/**
+ * @brief check address from a symbol located in a shared lilbrary
+ * @param hint text describing what to test
+ * @param libname libray to get the address from
+ * @param addrsym symbol to get the address for
+ * @param should_match result should match the given values
+ * @return 0 check passed
+ * @return 1 check failed
+ * @return 2 failed to open library
+ * @return 3 failed to get symbol address
+ */
+static int check_dladdr_by_dlopen( const char *hint, char *libname, char *sym, int should_match )
+{
+    void *library = NULL;
+    void *addr = NULL;
+    int result;
+
+    library = dlopen( libname, RTLD_GLOBAL );
+    if ( library == NULL )
+    {
+        fprintf( stderr, "could not open '%s'\n", libname );
+        return 2;
+    }
+
+    addr = dlsym( library, sym );
+    if ( !addr ) {
+        fprintf( stderr, "could not get address from library '%s' for symbol '%s'\n", libname, sym );
+        return 3;
+    }
+
+    result = check_dladdr( hint, addr, sym, should_match );
+    dlclose( library );
+
+    return result;
+}
+#endif
+
+#ifdef _WIN32
+#include <windows.h>
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+/* hide warning "reclared without 'dllimport' attribute" */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wattributes"
+#elif defined(_MSC_VER)
+/* disable warning C4273 inconsistent dll linkage */
+#pragma warning(push)
+#pragma warning(disable: 4273)
+#endif
+/*
+ * Windows API functions decorated with WINBASEAPI are imported from iat.
+ * For testing force linking by import thunk for the following functions
+ */
+HMODULE WINAPI GetModuleHandleA (LPCSTR lpModuleName);
+SIZE_T WINAPI VirtualQuery (LPCVOID lpAddress, PMEMORY_BASIC_INFORMATION lpBuffer, SIZE_T dwLength);
+/* force linking by iat */
+/* WINBASEAPI, which is normally used here, can be overriden by compiler or application, so we cannot use it */
+__declspec(dllimport) HMODULE WINAPI LoadLibraryExA(LPCSTR lpLibFileName, HANDLE hFile, DWORD  dwFlags);
+__declspec(dllimport) int __cdecl atoi(const char *_Str);
+
+#if defined(__GNUC__) && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 6))
+#pragma GCC diagnostic pop
+#elif defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+#endif
+
+#ifdef _WIN32
+#define PassWithDifferentAddressOnWin PassWithDifferentAddress
+#define FailOnWin Fail
+#else
+#define PassWithDifferentAddressOnWin Pass
+#define FailOnWin Pass
+#endif
+
+#if defined(_WIN32) && !defined(DLFCN_WIN32_SHARED)
+#define PassOnSharedBuild Fail
+#else
+#define PassOnSharedBuild Pass
+#endif
+
+#define UNUSED(x) (void)(x)
+
+int main(int argc, char **argv)
+{
+    UNUSED(argv);
+    unsigned char buffer[6] = { 0xFF, 0x25, 0x00, 0x00, 0x00, 0x00 };
```

Comment:
Variables must be declared prior code, so you cannot declare `unsigned char buffer[6]` after code `UNUSED(argv)`.

Due to this issue compilation under WDK is failing on error:
```
test-dladdr.c(153) : error C2143: syntax error : missing ';' before 'type'
```

Reply:
I changed the UNUSED macro similar to what is used by Qt, which should solve the issue.

Reply:
No, it does not solve anything. `(void)argv;` or `(void)(argv);` is still the same thing. It is a code and after it declaration of variables cannot be present.